### ISSUE TITLE
ResultSet not returning from successful procedure call when called using io.vertx.ext.sql.SQLClient

### ISSUE THREAD

**Issue by mikeluz** [vert-x3/vertx-jdbc-client#156]
### Version

* vert.x core: 3.5.3
* vert.x jdbc-client: 3.5.3
* jdbc driver: com.microsoft.sqlserver.jdbc.SQLServerDriver
* RDBMS: SQLServer

### Context

I have a SQLServer stored procedure that, when called using the Vert.X SQLClient (A) will not return a ResultSet (but does return db success code/return value so the db is hit and the proc does run) whereas when it is called using a standard Java ds (B) the ResultSet is returned:

A-Vert.x Jdbc Client (ResultSet is null)
    dbClient = dbClient.getConnection(res -> {
    connection.callWithParams(
      "{ ? = call procedureName(?,?) }",
      inputArray,
      outputArray,
      generateDbCallHandler(config, returnFuture)); // db success code of 0 received but ResultSet is null
    });

B-Standard Java SQL Datasource (ResultSet is NOT null)
        Connection conn =  DriverManager.getConnection(System.getenv(MainApplication.EnvVars.DB_URL) + ";databaseName=" + System.getenv(MainApplication.EnvVars.DB_NAME), System.getenv(MainApplication.EnvVars.DB_USERNAME), System.getenv(MainApplication.EnvVars.DB_PASSWORD));
        Statement statement = conn.createStatement();
        java.sql.ResultSet rs = statement.executeQuery(
          "EXEC " +
            "    procedureName" +
            "        @one_id = 578," +
            "        @two_id = 48");
        while(rs.next()){
          // rs is NOT null, am able to get the row this proc should return
        }

Any ideas on how to tweak A such that it returns the ResultSet it's supposed to (i.e., what I get in B)? A has been working for all of my procedures but this one -- which as I say works in B as well as when executed using a local SQLServer client.

### Do you have a reproducer?

* Issue is specific to the procedure which is private so not really able to offer a reproducer

Basically I'm trying to determine why this one procedure isn't returning a ResultSet when called using the Vert.X client whereas it does return when called using the basic Java SQL Connection method -- this seems like either a limitation on the Vert.X Jdbc Client OR I'm missing something regarding exactly how to use it in this case.

Any info appreciated!
